okhttp/src/main/java/okhttp3/internal/platform/android/Android10SocketAdapter.kt

```diff
@@ -27,16 +28,13 @@ import okhttp3.internal.platform.Platform
 /**
  * Simple non-reflection SocketAdapter for Android Q.
  */
+@SuppressLint("NewApi")
 class Android10SocketAdapter : SocketAdapter {
-  private val socketFactory =
-      SSLCertificateSocketFactory.getDefault(10000) as SSLCertificateSocketFactory
-
   override fun trustManager(sslSocketFactory: SSLSocketFactory): X509TrustManager? = null
 
   override fun matchesSocketFactory(sslSocketFactory: SSLSocketFactory): Boolean = false
 
-  override fun matchesSocket(sslSocket: SSLSocket): Boolean = sslSocket.javaClass.name.startsWith(
-      "com.android.org.conscrypt")
+  override fun matchesSocket(sslSocket: SSLSocket): Boolean = SSLSockets.isSupportedSocket(sslSocket)
```

Comment:
sweet!

Reply:
Agree, it's just kind of funny that we can't ignore this overall, just in one class at runtime on new devices.

And we have the horror that is DeferredSocketAdapter for dealing with classes that even trying reflection at runtime can fail fatally.  So we sample the socket as it goes past instead.

```
  @Synchronized private fun getDelegate(actualSSLSocketClass: SSLSocket): SocketAdapter? {
    if (!initialized) {
      try {
        var possibleClass: Class<in SSLSocket> = actualSSLSocketClass.javaClass
        while (possibleClass.name != "$socketPackage.OpenSSLSocketImpl") {
          possibleClass = possibleClass.superclass

          if (possibleClass == null) {
            throw AssertionError(
                "No OpenSSLSocketImpl superclass of socket of type $actualSSLSocketClass")
          }
        }

        delegate = AndroidSocketAdapter(possibleClass)
      } catch (e: Exception) {
        Platform.get()
            .log("Failed to initialize DeferredSocketAdapter $socketPackage", Platform.WARN, e)
      }

      initialized = true
    }

    return delegate
  }
```